### DISCUSSION TITLE
Lazy instanciation of ResourceDependency

### DISCUSSION THREAD

**Discussion by geoHeil** [dagster-io/dagster#18312] [Q&A] [answered]
How to use

pyspark: ResourceDependency[PySparkResource]

for lazy instanciation?

Assuming an IO manger like https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/parquet_io_manager.py#L17 which supports pandas and pyspark mode - how can I initialize the pyspark object only lazily?

**Comment by smackesey** [comment]
I don't think this is currently supported, looks like we are already tracking it here: https://github.com/dagster-io/dagster/issues/7585

**Reply by geoHeil** [reply]
it looks like there is a `LazyPySparkResource`? Can you share an exampled how it should be used wit hthe new pythonic API inside  `ResourceDependency`?

**Reply by smackesey** [reply, marked as answer]
My mistake, you're right: https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-pyspark/dagster_pyspark/resources.py#L94-L140

Looks like this is just a regular `ConfigurableResource`, so you should be able to do:

`pyspark: ResourceDependency[LazyPySparkResource]`
